File: TC_pruebas_login/test_puppetter/tc_03.js
const puppeteer = require('puppeteer');

// Variables
const link = "https://the-internet.herokuapp.com/login";
const uname_input = '#username';
const pw_input = '#password';
const btn_continue = '#login button i';

let counter = 1;

async function tomar_screenshot(page, screen_name) {
    await page.screenshot({
        path: `test_images/tc_03_${screen_name}_${counter}.png`
    });
    counter++;
}

async function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

(async () => {
    const browser = await puppeteer.launch({ headless: false });
    const page = await browser.newPage();

    await page.goto(link, { waitUntil: 'networkidle2' });
    await sleep(3000);
    await tomar_screenshot(page, "inicio_vacio");

    // Username vacío
    await page.type(uname_input, "");

    // Password vacío
    await page.type(pw_input, "");
    await tomar_screenshot(page, "campos_vacios");

    // Clic en login
    await page.click(btn_continue);
    await sleep(2000);

    try {
        const error_message = await page.$('#flash');
        if (error_message) {
            const error_text = await page.evaluate(el => el.textContent, error_message);
            await tomar_screenshot(page, "resultado");

            if (error_text.includes("Campos obligatorios")) {
                console.log(`✅ Test exitoso: ${error_text}`);
            } else {
                console.log("❌ Test fallido: Mensaje de error no encontrado");
            }
        } else {
            console.log("❌ Test fallido: No se encontró el mensaje de error");
        }
    } catch (e) {
        console.log(`❌ Test fallido: Ocurrió un error - ${e}`);
    }

    await browser.close();
})();

File: TC_pruebas_login/test_playwright/tc_03.py
from playwright.sync_api import sync_playwright
from var import *

counter = 1

def tomar_screenshot(page, screen_name: str):
    global counter
    page.screenshot(path=f"TC_pruebas_login/test_playwright/test_images/tc_03_{screen_name}_{counter}.png")
    counter += 1

with sync_playwright() as p:
    browser = p.chromium.launch(headless=False)
    page = browser.new_page()

    page.goto(link)
    page.wait_for_timeout(3000)
    tomar_screenshot(page, "inicio_vacio")

    page.locator(uname_input).fill("")
    page.locator(pw_input).fill("")
    tomar_screenshot(page, "campos_vacios")

    page.locator(btn_continue).click()
    page.wait_for_timeout(2000)

    try:
        error_message = page.locator('//*[@id="flash"]')
        error_text = error_message.text_content()
        tomar_screenshot(page, "resultado")

        if "Campos obligatorios" in error_text:
            print(f"Test exitoso: {error_text}")
        else:
            print("Test fallido: Mensaje de error no encontrado")
    except Exception as e:
        print(f"Test fallido: Ocurrió un error - {e}")

    browser.close()
